discovery/wh_discovery.go

```diff
@@ -0,0 +1,162 @@
+package discovery
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"sync"
+	"time"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/livepeer/go-livepeer/core"
+	"github.com/livepeer/go-livepeer/net"
+
+	"github.com/golang/glog"
+)
+
+var whRefreshInterval = 1 * time.Minute
+
+type webhookResponse struct {
+	Address string
+}
+
+type webhookPool struct {
+	node         *core.LivepeerNode
+	pool         *orchestratorPool
+	callback     *url.URL
+	responseHash ethcommon.Hash
+	lastRequest  time.Time
+	mu           *sync.RWMutex
+}
+
+func NewWebhookPool(node *core.LivepeerNode, callback *url.URL) *webhookPool {
```

Comment:
The repetition here can probably be avoided, would this work?

```golang
func NewWebhookPool(node *core.LivepeerNode, callback *url.URL) *webhookPool {
  p := &webhookPool {
    node: node,
    callback: callback
    mu: &sync.RWMutex{}
  }
  go p.getURLs()
  return p
}
```

Reply:
Much more concise - thank you